src/utils/staking/index.ts

```diff
@@ -43,3 +63,133 @@ export const buildStakingOutput = (
+    termWithFixed = term;
+  }
+  return termWithFixed;
+};
+
+/**
+ * Validate the staking transaction input data.
+ *
+ * @param {number} stakingAmountSat - The staking amount in satoshis.
+ * @param {number} stakingTerm - The staking term in blocks.
+ * @param {Params} params - The staking parameters.
+ * @param {UTXO[]} inputUTXOs - The input UTXOs.
+ * @param {number} feeRate - The Bitcoin fee rate in sat/vbyte
+ * @throws {StakingError} - If the input data is invalid.
+ */
+export const validateStakingTxInputData = (
+  stakingAmountSat: number,
+  stakingTerm: number,
+  params: Params,
+  inputUTXOs: UTXO[],
+  feeRate: number,
+) => {
+  if (
+    stakingAmountSat < params.minStakingAmountSat ||
+    stakingAmountSat > params.maxStakingAmountSat
+  ) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "Invalid staking amount",
+    );
+  }
+
+  if (
+    stakingTerm < params.minStakingTimeBlocks ||
+    stakingTerm > params.maxStakingTimeBlocks
+  ) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "Invalid staking term",
+    );
+  }
+
+  if (inputUTXOs.length == 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "No input UTXOs provided",
+    );
+  } else if (feeRate <= 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_INPUT, "Invalid fee rate",
+    );
+  }
+}
+
+/**
+ * Validate the parameters for staking.
+ * 
+ * @param {Params} params - The staking parameters.
+ * @throws {StakingError} - If the parameters are invalid.
+ */
+export const validateParams = (params: Params) => {
+  if (params.covenantPks.length == 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Could not find any covenant public keys",
+    );
+  }
+  if (params.covenantPks.length < params.covenantQuorum) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Covenant public keys must be greater than or equal to the quorum",
+    );
+  }
+  if (params.unbondingTime <= 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Unbonding time must be greater than 0",
+    );
+  }
+  if (params.unbondingFeeSat <= 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Unbonding fee must be greater than 0",
+    );
+  }
+  if (params.maxStakingAmountSat < params.minStakingAmountSat) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Max staking amount must be greater or equal to min staking amount",
+    );
+  }
+  if (params.minStakingAmountSat < 0) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Min staking amount must be greater than 0",
+    );
+  }
+  if (params.maxStakingTimeBlocks < params.minStakingTimeBlocks) {
+    throw new StakingError(
+      StakingErrorCode.INVALID_PARAMS,
+      "Max staking time must be greater or equal to min staking time",
+    );
+  }
+  if (params.minStakingTimeBlocks < 0) {
```

Comment:
`params.minStakingTimeBlocks < 0` -> `params.minStakingTimeBlocks <= 0` ? Errors message states that `Max staking time must be greater than 0`

Reply:
humh I would say imake no sense to have `minStakingTimeBlocks=0` therefore we need to throw error when `params.minStakingTimeBlocks <= 0` ? 😅  do I miss something here?